Scripts/RemediationScripts/Remediate-EnableImmutabilitySettingForRecoveryServiceVault.ps1

```diff
@@ -0,0 +1,551 @@
+
+            # Get Recovery Services Vault(s) properties
+            $RecoveryServiceVaultDetailsForSubscription | ForEach-Object {                
+    
+                try {                
+                    $RecoveryServiceVaultResource = Get-AzRecoveryServicesVault -ResourceGroupName $_.ResourceGroupName -Name $_.Name -ErrorAction SilentlyContinue
+                    $RecoveryServiceVaultDetails += Get-RecoveryServiceVaultDetails $RecoveryServiceVaultResource
+                }
+                catch {
+                    Write-Host "Error fetching Recovery Services Vault(s) resource: Resource ID:  [$($resourceId)]. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                }
+            
+            }
+        }
+        else {
+            if (-not (Test-Path -Path $FilePath)) {
+                Write-Host "ERROR: Input file: [$($FilePath)] not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host $([Constants]::DoubleDashLine)	
+                return
+            }
+            Write-Host "Fetching all Recovery Services Vault(s) from [$($FilePath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            $RecoveryServiceVaults = Import-Csv -LiteralPath $FilePath
+    
+            $validRecoveryServiceVaults = $RecoveryServiceVaults | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+    
+            $validRecoveryServiceVaults | ForEach-Object {
+                $resourceId = $_.ResourceId
+    
+                try {                
+                    $RecoveryServiceVaultResource = Get-AzRecoveryServicesVault -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
+                    $RecoveryServiceVaultDetails += Get-RecoveryServiceVaultDetails $RecoveryServiceVaultResource
+                }
+                catch {
+                    Write-Host "Error fetching Recovery Services Vault(s) resource: Resource ID:  [$($resourceId)]. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                }
+            }
+        }
+    }
+
+    $totalRecoveryServiceVaults = ($RecoveryServiceVaultDetails | Measure-Object).Count
+
+    if ($totalRecoveryServiceVaults -eq 0) {
+        Write-Host "No Recovery Services Vault(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)	
+        return
+    }
+
+    Write-Host "Found [$($totalRecoveryServiceVaults)] Recovery Services Vault(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # Recovery Services Vault(s) where immutability is not locked
+    $RecoveryServiceVaultsWithoutImmutability = @()
+
+    Write-Host "Separating Recovery Services Vault(s) for which immutability is not locked..." -ForegroundColor $([Constants]::MessageType.Info)
```

Comment:
We are communicating just about "not locked" state, but control would look for both enabled and locked right. Please tweak the messaging accordingly.

Reply:
Done